python/lsst/jointcal/jointcal.py

```diff
@@ -419,32 +503,260 @@ class JointcalTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+                                                      tract=tract,
+                                                      reject_bad_fluxes=True)
+            photometry_output = self._make_output(associations.getCcdImageList(),
+                                                  photometry.model,
+                                                  "toPhotoCalib")
+        else:
+            photometry_output = None
+
+        return pipeBase.Struct(outputWcs=astrometry_output,
+                               outputPhotoCalib=photometry_output,
+                               job=self.job,
+                               astrometryRefObjLoader=self.astrometryRefObjLoader,
+                               photometryRefObjLoader=self.photometryRefObjLoader)
+
+    def _make_schema_table(self):
+        """Return an afw SourceTable that to use as a base for creating the
+        SourceCatalog to insert values from the dataFrame into.
+
+        Returns
+        -------
+        table : `lsst.afw.table.SourceTable`
+            Table with schema and slots to use to make SourceCatalogs.
+        """
+        schema = lsst.afw.table.SourceTable.makeMinimalSchema()
+        schema.addField("centroid_x", "D")
+        schema.addField("centroid_y", "D")
+        schema.addField("centroid_xErr", "F")
+        schema.addField("centroid_yErr", "F")
+        schema.addField("shape_xx", "D")
+        schema.addField("shape_yy", "D")
+        schema.addField("shape_xy", "D")
+        schema.addField("flux_instFlux", "D")
+        schema.addField("flux_instFluxErr", "D")
+        table = lsst.afw.table.SourceTable.make(schema)
+        table.defineCentroid("centroid")
+        table.defineShape("shape")
+        table.defineCalibFlux("flux")
+        return table
+
+    def _extract_detector_catalog_from_visit_catalog(self, table, visitCatalog, detectorId):
+        """Return an afw SourceCatalog extracted from a visit-level dataframe,
+        limited to just one detector.
+
+        Parameters
+        ----------
+        table : `lsst.afw.table.SourceTable`
+            Table factory to use to make the Catalog to be filled in.
+        visitCatalog : `pandas.DataFrame`
+            DataFrame to extract a detector catalog from.
+        detectorId : `int`
+            Numeric id of the detector to extract from ``visitCatalog``.
+
+        Returns
+        -------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Detector-level catalog extracted from ``visitCatalog``.
+        """
+        # map from dataFrame column to afw table column
+        mapping = {'sourceId': 'id',
+                   'x': 'centroid_x',
+                   'y': 'centroid_y',
+                   'xErr': 'centroid_xErr',
+                   'yErr': 'centroid_yErr',
+                   'Ixx': 'shape_xx',
+                   'Iyy': 'shape_yy',
+                   'Ixy': 'shape_xy',
+                   # TODO: "base_CircularApertureFlux" gets renamed to "ApFlux",
```

Comment:
Could you make a ticket for this TODO, and reference the ticket number here?

Reply:
Eli helped me work out a fix, so this TODO is gone.